client/src/observatory/results.tsx

```diff
@@ -0,0 +1,668 @@
+          background: `conic-gradient(var(--button-color) 0grad, ${progressPercent}%, rgba(0,0,0,0) ${progressPercent}% 100%)`,
+        }}
+      ></div>
+      <small>
+        Wait {remainingSecs}s to {title.toLowerCase()}
+      </small>
+    </Button>
+  ) : (
+    <Button onClickHandler={rescan}>{title}</Button>
+  );
+}
+
+function ObservatoryTests({ result }: { result: ObservatoryResult }) {
+  return Object.keys(result.tests).length !== 0 ? (
+    <section className="tab-content">
+      <figure className="scroll-container">
+        <table className="tests">
+          <thead>
+            <tr>
+              <th>Test</th>
+              <th>Score</th>
+              <th>Reason</th>
+              <th>Recommendation</th>
+            </tr>
+          </thead>
+          <tbody>
+            {Object.entries(result.tests).map(([name, test]) => {
+              return (
+                <tr key={name}>
+                  <td data-header="Test">
+                    <Link href={test.link}>{test.title}</Link>
+                  </td>
+                  {test.pass === null ? (
+                    <td data-header="Score">-</td>
+                  ) : (
+                    <td className="score" data-header="Score">
+                      <span>
+                        <span className="obs-score-value">
+                          {fixMinusSymbol(`${test.score_modifier}`)}
+                        </span>
+                        <PassIcon pass={test.pass} />
+                      </span>
+                    </td>
+                  )}
+                  <td
+                    data-header="Reason"
+                    dangerouslySetInnerHTML={{
+                      __html: test.score_description,
+                    }}
+                  />
+                  <td
+                    data-header="Advice"
+                    dangerouslySetInnerHTML={{
+                      __html:
+                        test.recommendation || `<p class="obs-none">None</p>`,
+                    }}
+                  />
+                </tr>
+              );
+            })}
+          </tbody>
+        </table>
+      </figure>
+    </section>
+  ) : null;
+}
+
+function ObservatoryHistory({ result }: { result: ObservatoryResult }) {
+  return result.history.length ? (
+    <section className="tab-content">
+      <figure className="scroll-container">
+        <table className="history">
+          <thead>
+            <tr>
+              <th>Date</th>
+              <th>Score</th>
+              <th>Grade</th>
+            </tr>
+          </thead>
+          <tbody>
+            {[...result.history]
+              .reverse()
+              .map(({ scanned_at, score, grade }) => (
+                <tr key={scanned_at}>
+                  <td data-header="Date">
+                    {new Date(scanned_at).toLocaleString([], {
+                      dateStyle: "full",
+                      timeStyle: "medium",
+                    })}
```

Comment:
As mentioned above, this could be a helper function like `formatDateTime(dateStyle = "medium", timeStyle = "medium")`.

Reply:
done.